src/components/common/ContactForm/index.jsx

```diff
@@ -47,7 +48,8 @@ class ContactForm extends React.Component {
       [name]: value,
     };
     this.setState({ data });
-    this.validate(data);
+    const { errors } = validate(schema, data);
+    this.setState({ errors });
```

Comment:
I don't see any reasons to update `data` before calling `validate`, so you can just combine `data` and `errors` update into 1 `setState`
```suggestion
    this.setState({ data, errors });
```

Reply:
I also see the same logic in other places, so please update them too.